### DISCUSSION TITLE
SSLKEYLOGFILE support?

### DISCUSSION THREAD

**Discussion by cslev** [DNSCrypt/dnscrypt-proxy#2401] [Ideas]
Hi,
I am amazed about this project and highly appreciate the authors' efforts in making such a great tool.
For network traffic analysis, I was wondering whether dnscrypt-proxy supports storing SSL keys used for establishing a connection to resolvers (e.g., DoH resolvers).
I want to capture the traffic dnscrypt-proxy generates, use the SSL keys to decrypt the traffic (e.g., in Wireshark) and, among other things, observe which DNS query was sent to which resolver.
Note, I am aware of the built-in logging features, I just want to see everything in an as is basis on the wire.

I tried the typical linux way of adding SSLKEYLOGFILE ENV variable in front of the dnscrypt-proxy command, but it does not work.

Thanks

**Comment by jedisct1** [comment]
Sure!

Done, in 0c26d1637a8628425476f09c6f27bdbc15116d06

Thanks for suggesting this!

**Reply by cslev** [reply]
Wow, amazingly blazingly fast :) 
Thank you
